src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIPacketPool.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace Microsoft.Data.SqlClient.SNI
+{
+    // this is a very simple threadsafe pool derived from the aspnet/extensions default pool implementation
+    // https://github.com/aspnet/Extensions/blob/master/src/ObjectPool/src/DefaultObjectPool.cs
```

Comment:
Link leads to 404 error.

Reply:
Things have recently been reshuffled in dotnet. Here's a better link:
https://github.com/dotnet/extensions/blob/release/3.1/src/ObjectPool/src/DefaultObjectPool.cs